### PR TITLE
Simplify BackedEnum

### PR DESCRIPTION
**Pull request by szepeviktor**
To #21

### CODE REVIEW

**Comment by PrinsFrank** [comment]
@szepeviktor Sorry, I see why you wanted this PR to be merged before the other branch was merged. Can you open a new PR to main?

**Comment by szepeviktor** [comment]
Changed base.

**Comment by szepeviktor** [comment]
@PrinsFrank Astronomically sad news 😢 😢 😢

```
Frank's result: 2.1294710636139
Viktor's result: 0.045840978622437
```

**Comment by szepeviktor** [comment]
```php
function FrankVehicle(string $fqn, string $keyName): ?\BackedEnum
{
    foreach ($fqn::cases() as $case) {
        if ($case->name === $keyName) {
            return $case;
        }
    }

    return null;
}

function ViktorVehicle(string $fqn, string $keyName): ?\BackedEnum
{
    return defined("{$fqn}::{$keyName}") ? constant("{$fqn}::{$keyName}") : null;
}

$frankslap = microtime(true);
for ($i = 0; $i < 10**5; $i += 1) {
    FrankVehicle('ISO639_2_Alpha_3_Common', 'Zuni');
}
echo "Frank's result: ", microtime(true) - $frankslap, PHP_EOL;

$viktorslap = microtime(true);
for ($i = 0; $i < 10**5; $i += 1) {
    ViktorVehicle('ISO639_2_Alpha_3_Common', 'Zuni');
}
echo "Viktor's result: ", microtime(true) - $viktorslap, PHP_EOL;
```

**Comment by szepeviktor** [comment]
Finding the first enum:

```
Frank's result: 0.94744300842285
Viktor's result: 0.053514003753662
```

**Comment by PrinsFrank** [comment]
@szepeviktor Wow, that is not sad news but great news! That's a 98% decrease!

File: src/BackedEnum.php [hunk 1]
```diff
@@ -23,12 +23,6 @@ public static function fromKey(string $fqn, string $keyName): \BackedEnum
...
     public static function tryFromKey(string $fqn, string $keyName): ?\BackedEnum
     {
-        foreach ($fqn::cases() as $case) {
-            if ($case->name === $keyName) {
-                return $case;
-            }
-        }
-
-        return null;
+        return defined("{$fqn}::{$keyName}") ? constant("{$fqn}::{$keyName}") : null;
```
Comment (PrinsFrank) [inline src/BackedEnum.php:26]: One small thing: these curly brackets are not necessary:
```
return defined("$fqn::$keyName") ? constant("$fqn::$keyName") : null;
```

Great that you found such an elegant oneliner that is so much quicker!

Comment (szepeviktor) [reply src/BackedEnum.php:26]: My heart hearts. Those braces are for readability. Without those humans are not able to parse the string.

Comment (szepeviktor) [reply src/BackedEnum.php:26]:
```suggestion
        return defined("$fqn::$keyName") ? constant("$fqn::$keyName") : null;
```

**Comment by PrinsFrank** [comment]
That's what you get when you set PHPStan to level 9. Thanks a lot, This is so much better than iterating over all keys!

**Pull request merged**